cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -407,21 +407,24 @@ void SparsePoseGraph::FreezeTrajectory(const int trajectory_id) {
   });
 }
 
-void SparsePoseGraph::AddSubmapFromProto(const int trajectory_id,
-                                         const transform::Rigid3d& initial_pose,
+void SparsePoseGraph::AddSubmapFromProto(const transform::Rigid3d& pose,
                                          const mapping::proto::Submap& submap) {
   if (!submap.has_submap_2d()) {
     return;
   }
 
+  const mapping::SubmapId proto_submap_id = {submap.submap_id().trajectory_id(),
+                                             submap.submap_id().submap_index()};
+  const int trajectory_id = proto_submap_id.trajectory_id;
   std::shared_ptr<const Submap> submap_ptr =
       std::make_shared<const Submap>(submap.submap_2d());
-  const transform::Rigid2d initial_pose_2d = transform::Project2D(initial_pose);
+  const transform::Rigid2d pose_2d = transform::Project2D(pose);
 
   common::MutexLocker locker(&mutex_);
   trajectory_connectivity_state_.Add(trajectory_id);
   const mapping::SubmapId submap_id =
       submap_data_.Append(trajectory_id, SubmapData());
+  CHECK_EQ(submap_id, proto_submap_id);
```

Comment:
This PR removes support for serialized data with trimmed submaps. Add a `TODO` that this does not support trimmed submaps yet and add output reflecting this to the `CHECK`? Similar for 3D.

Reply:
I had the impression that, while serializing, IDs are remapped to look like there wasn't any trimming at all. Isn't this the case?

Reply:
Now I see that in `MapBuilder`, you [do not apply the same remapping](https://github.com/googlecartographer/cartographer/blob/cae9c02e0589fb3d7ae16372c9803a0f47b3829f/cartographer/mapping/map_builder.cc#L161), but you have left it as a TODO. When the TODO is completed, this `CHECK` will completely be fine, right?